### PR TITLE
CEDS-2563 Add eori whitelisting

### PR DESCRIPTION
**Pull request by PatrykRudnicki**

### CODE REVIEW

File: app/controllers/actions/AuthAction.scala [hunk 1]
```diff
@@ -67,6 +73,22 @@ class AuthActionImpl @Inject()(val authConnector: AuthConnector, val config: Con
...
+
+    eoriOpt.exists(eori => eoriWhitelist.allows(eori.value))
+  }
 }
 
 trait AuthAction extends ActionBuilder[AuthenticatedRequest, AnyContent] with ActionRefiner[Request, AuthenticatedRequest]
+
+@ProvidedBy(classOf[EoriWhitelistProvider])
+class EoriWhitelist(val values: Seq[String]) {
+  def allows(eori: String): Boolean = values.isEmpty || values.contains(eori)
```
Comment (SteveSugden) [inline app/controllers/actions/AuthAction.scala:88]: Is that right?  No white list means everyone is allowed?

Comment (PatrykRudnicki) [reply app/controllers/actions/AuthAction.scala:88]: Yes, the same approach is in Exports service

File: test/controllers/actions/AuthActionSpec.scala [hunk 2]
```diff
@@ -77,14 +77,78 @@ class AuthActionSpec extends ControllerSpecBase {
...
         redirectLocation(response) mustBe Some(controllers.routes.UnauthorisedController.onPageLoad.url)
       }
     }
   }
 
+  "Auth action" should {
+
+    "allow access" when {
+
+      "whitelisting doesn't have any eori and user has eori" in {
```
Comment (SteveSugden) [inline test/controllers/actions/AuthActionSpec.scala:92]: i see it is correct!

Comment (SteveSugden) [reply test/controllers/actions/AuthActionSpec.scala:92]: Actually, i should have said, "I see you meant to do it like that!"  :-)

I guess once a service is "live" we remove the whitelist and then it's free-for-all.

Comment (PatrykRudnicki) [reply test/controllers/actions/AuthActionSpec.scala:92]: yeah, that's true :)

File: app/controllers/actions/AuthAction.scala [hunk 3]
```diff
@@ -67,6 +73,22 @@ class AuthActionImpl @Inject()(val authConnector: AuthConnector, val config: Con
...
+  private def isEoriWhitelisted(enrolments: Enrolments): Boolean = {
+    val eoriOpt = enrolments.getEnrolment("HMRC-CUS-ORG").flatMap(_.getIdentifier("EORINumber"))
+
+    eoriOpt.exists(eori => eoriWhitelist.allows(eori.value))
+  }
 }
 
 trait AuthAction extends ActionBuilder[AuthenticatedRequest, AnyContent] with ActionRefiner[Request, AuthenticatedRequest]
+
+@ProvidedBy(classOf[EoriWhitelistProvider])
```
Comment (SteveSugden) [inline app/controllers/actions/AuthAction.scala:86]: This google ProvidedBy thingy seems a bit over-kill.   You are injecting the config to get the list of white-listed Eoris.    Why not build your EoriWhitelist class using the config?

Comment (PatrykRudnicki) [reply app/controllers/actions/AuthAction.scala:86]: Hmm, I copied this from Exports service

I can implement it in different way if you want :)

Comment (SteveSugden) [reply app/controllers/actions/AuthAction.scala:86]: Up to you.  Importing google.anything always makes me nervous for some reason.

**Comment by PatrykRudnicki** [comment]
Yes, I agree with you Steve, I'll merge it, because it's necessary for the trial, but I'll prepare a separate PR with simplified implementation

**Pull request merged**